src/language-server/project/rover/project.ts

```diff
@@ -0,0 +1,190 @@
+import { ProjectStats } from "src/messages";
+import { DocumentUri, GraphQLProject } from "../base";
+import { TextDocument } from "vscode-languageserver-textdocument";
+import { generateKeyBetween } from "fractional-indexing";
+import {
+  CancellationToken,
+  SymbolInformation,
+  InitializeRequest,
+  StreamMessageReader,
+  StreamMessageWriter,
+  createProtocolConnection,
+  ProtocolConnection,
+  ClientCapabilities,
+  ProtocolRequestType,
+  CompletionRequest,
+  ProtocolNotificationType,
+  DidChangeWatchedFilesNotification,
+  HoverRequest,
+  CancellationTokenSource,
+} from "vscode-languageserver/node";
+import cp from "node:child_process";
+import { GraphQLProjectConfig } from "../base";
+import { ApolloConfig, RoverConfig } from "../../config";
+import { DocumentSynchronization } from "./DocumentSynchronization";
+
+export function isRoverConfig(config: ApolloConfig): config is RoverConfig {
+  return config instanceof RoverConfig;
+}
+
+export interface RoverProjectConfig extends GraphQLProjectConfig {
+  config: RoverConfig;
+  capabilities: ClientCapabilities;
+}
+
+export class RoverProject extends GraphQLProject {
+  config: RoverConfig;
+  _connection?: Promise<ProtocolConnection>;
+  capabilities: ClientCapabilities;
+  get displayName(): string {
+    return "Rover Project";
+  }
+  private documents = new DocumentSynchronization(
+    this.sendNotification.bind(this),
+  );
+
+  constructor(options: RoverProjectConfig) {
+    super(options);
+    this.config = options.config;
+    this.capabilities = options.capabilities;
+  }
+
+  initialize() {
+    return [this.connection.then(() => {})];
+  }
+
+  get connection(): Promise<ProtocolConnection> {
```

Comment:
I'm curious, is this used by the superclass at all? If not, I tend to prefer `async getConnection()` here and make this an actual method, mostly because I think it helps a touch when reading it in lines like this:

```ts
const connection = await this.getConnection();
```

Without making it a function, it looks both redundant and a bit confusing why we await the property:

```ts
const connection = await this.connection;
```

I'm happy to leave this as-is too, just figured this might make it a touch clearer whats happening here.

Reply:
I like the suggestion, but I'll do it in the next PR :)